### PR TITLE
systemd: 246.6 -> 247

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@flokli @arianvp feel free to push on top. Build is fixed but I have not booted from it yet.

**Comment by Mic92** [comment]
I upstreamed https://github.com/systemd/systemd/pull/17580 which means a big patch less between nixos-systemd and upstream. This is included in rc2.

**Comment by arianvp** [comment]
We should make sure all our udev rules of all our packages are compatible before we can merge this. Due to the breaking changes in the handling of bind and unbind events.

**Comment by flokli** [comment]
> […]  all our udev rules of all our packages are compatible before we can merge this […]

I'd say that's a bit ambitions, if not unrealistic, as this is all manual testing - or is there a easy search pattern to spot these incompatibilities?

Before merging, we should do a much more thorough testing of all the GUI VM tests, to spot regressions (and it'd help to have a Hydra jobset for this, so more people can test), but "Unstable" is the right place to track some of these regressions in less-popular components.

File: pkgs/os-specific/linux/systemd/default.nix [hunk 1]
```diff
@@ -70,6 +71,7 @@
 , withMachined ? true
 , withNetworkd ? true
 , withNss ? true
+, withOomd ? true
```
Comment (flokli) [inline pkgs/os-specific/linux/systemd/default.nix:74]: is there anything in here that users need to configure? Can we test it being present in a VM test?

Comment (Mic92) [reply pkgs/os-specific/linux/systemd/default.nix:74]: It requires a service
example/systemd/system/systemd-oomd.service
and some dbus stuff. Right now its not enabled.
share/dbus-1/system.d/org.freedesktop.oom1.conf

Comment (Mic92) [reply pkgs/os-specific/linux/systemd/default.nix:74]: And it wants swap: 

```
sudo /nix/store/0y9z2b5c245y0gkj1cw0q212963421dy-systemd-247-rc2/lib/systemd/systemd-oomd
Requires swap to operate
```

Comment (Mic92) [reply pkgs/os-specific/linux/systemd/default.nix:74]: We can also disable it by default until someones wants to properly integrate it?

Comment (flokli) [reply pkgs/os-specific/linux/systemd/default.nix:74]: yes, I think we should only enable things which we have integration tests for. We can keep the flag, so it's easier for people to enable it when they want to start working on it (like we did with homed)

**Comment by flokli** [comment]
247 was released, I updated this PR to drop https://github.com/systemd/systemd/pull/17608 included in the final release, and reflowed the patches once more.

**Comment by flokli** [comment]
This needed another rebase apparently.

@Mic92 what's your take on this? Should we let this cook in staging a bit? I ran a bunch of tests, but can't rebuild the world ;-)

**Comment by Mic92** [comment]
> This needed another rebase apparently.
> 
> @Mic92 what's your take on this? Should we let this cook in staging a bit? I ran a bunch of tests, but can't rebuild the world ;-)

I am using this on my laptop for a while now.

**Comment by TredwellGit** [comment]
https://github.com/NixOS/nixpkgs/pull/105850#issuecomment-739226257
Reverting https://github.com/NixOS/nixpkgs/commit/ef050bc3d18d4c72da4ba238fccbf0d82565eeb9 unbreaks the packages.

**Comment by andersk** [comment]
This seems to have broken the channel-blocking networking.networkd.privacy test: #106858.

**Comment by FRidh** [comment]
https://github.com/NixOS/nixpkgs/issues/106858
> Friday evening I intend to revert the systemd upgrade unless there is a tested fix before then.

**Pull request merged**